core/src/main/scala/cats/data/NonEmptyChain.scala

```diff
@@ -22,14 +22,15 @@
 package cats
 package data
 
-import NonEmptyChainImpl.create
-
 import cats.instances.StaticMethods
 import cats.kernel._
 
 import scala.collection.immutable.SortedMap
 
-private[data] object NonEmptyChainImpl extends NonEmptyChainInstances with ScalaVersionSpecificNonEmptyChainImpl {
+// For internal purposes only - do not use!
+// This object is kept public for the sake of binary compatibility.
+// The visibility can be changed in future versions.
+object NonEmptyChainImpl extends NonEmptyChainInstances with ScalaVersionSpecificNonEmptyChainImpl {
```

Comment:
Maybe we can make this a real scaladoc comment `/** */`. And also, maybe mark these as deprecated to discourage accidental use? We can absolutely fix this in Cats 3.

Reply:
I tried making them `@deprecated` but seems it makes all their methods deprecated as well even if they're accessed via their corresponding `val` references. I'll give it another try, maybe I missed something out.

Reply:
Oh darn. I guess this was one advantage of your `abstract class` method 🤔

Even if we can't deprecate it in source, at least we can add a `@deprecated` tag inside the scaladoc comment.